benchmarking/evaluations/metrics/cosine_similarity.py

```diff
@@ -0,0 +1,128 @@
+import itertools
+import statistics
+from typing import List, Dict, Union, Tuple
+
+from numpy import dot
+from numpy.linalg import norm
+
+from api.models.student import Student
+from api.models.team import Team
+from api.models.team_set import TeamSet
+from benchmarking.evaluations.interfaces import TeamSetMetric
+
+
+class ClassAttributeSet:
+    def __init__(self, team_set: TeamSet):
+        # Dict of a (Attribute type, attribute value) to index of that combination in the vector used for cosine similarity
```

Comment:
the first thing is the attribute id right? super nit though, only fix if I find something else to actually request changes for